src/PromiseUtil.ts

```diff
@@ -1,9 +1,17 @@
 import { Logger } from "matrix-appservice-bridge";
+import { MatrixError } from "matrix-bot-sdk";
 
 const SLEEP_TIME_MS = 250;
 const DEFAULT_RETRY = () => true;
 const log = new Logger("PromiseUtil");
 
+export function retryMatrixErrorFilter(err: unknown) {
+    if (err instanceof MatrixError && err.statusCode >= 400 && err.statusCode <= 499) {
```

Comment:
It will retry, if the error isn't a MatrixError. Is that what we want?
I thought this is secifically for retrying on (temporary) server errors and rate limiting.

Are gateway timeouts MatrixErrors?

Reply:
The intention is to retry non-fatal errors, i.e. we will retry 5XX errors because more often than not it's the server blowing up. Anything that's not a MatrixError is also considered temporary (we err on the side of caution).